crates/sui-framework/packages/deepbook/sources/clob_v2.move

```diff
@@ -1274,6 +1502,7 @@ module deepbook::clob_v2 {
         let original_quantity = quantity;
         let base_quantity_filled;
         let quote_quantity_filled;
+        let meta_data: vector<MatchedOrderMetadata<BaseAsset, QuoteAsset>>;
```

Comment:
I suppose you could also have 
`let meta_data = if (is_bid) {...; matched_order_metadata } else { ...; matched_order_metadata }`
but up to you

Reply:
thats a good suggestion, returning metadata now from the if else block directly.